app/models/ta.rb

```diff
@@ -87,4 +89,10 @@ def grade_distribution_array(assignment, intervals = 20)
 
     return distribution
   end
+
+  private
+
+  def create_grader_permission
+    self.grader_permission = GraderPermission.create!(user_id: self.id)
```

Comment:
if you change this to a `before_validation` callback you should also update this to only build the GraderPermission object (and do it only if it doesn't exist so you don't overwrite a previously built one):

```
self.grader_permission ||= GraderPermission.new
```